Stripe/STPBankAccount.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  STPBankAccount.swift
+//  Stripe
+//
+//  Created by Charles Scalesse on 10/1/14.
+//
+//
+
+import Foundation
+
+/// Possible validation states for a bank account.
+@objc public enum STPBankAccountStatus: Int {
+  /// The account has had no activity or validation performed
+  case new
+  /// Stripe has determined this bank account exists.
+  case validated
+  /// Bank account verification has succeeded.
+  case verified
+  /// Verification for this bank account has failed.
+  case verificationFailed
+  /// A transfer sent to this bank account has failed.
+  case errored
+}
+
+/// Representation of a user's bank account details that have been tokenized with
+/// the Stripe API.
+/// - seealso: https://stripe.com/docs/api#bank_accounts
+public class STPBankAccount: NSObject, STPAPIResponseDecodable, STPSourceProtocol {
+  /// You cannot directly instantiate an `STPBankAccount`. You should only use one
+  /// that has been returned from an `STPAPIClient` callback.
+  required override init() {
+  }
+
+  /// The routing number for the bank account. This should be the ACH routing number,
+  /// not the wire routing number.
+  @objc public private(set) var routingNumber: String?
+  /// Two-letter ISO code representing the country the bank account is located in.
+  @objc public private(set) var country: String?
+  /// The default currency for the bank account.
+  @objc public private(set) var currency: String?
+  /// The last 4 digits of the account number.
+  @objc public private(set) var last4: String?
+  /// The name of the bank that owns the account.
+  @objc public private(set) var bankName: String?
+  /// The name of the person or business that owns the bank account.
+  @objc public private(set) var accountHolderName: String?
+  /// The type of entity that holds the account.
+  @objc public private(set) var accountHolderType: STPBankAccountHolderType = .individual
+  /// A proxy for the account number, this uniquely identifies the account and can be
+  /// used to compare equality of different bank accounts.
+  @objc public private(set) var fingerprint: String?
+  /// The validation status of the bank account. - seealso: STPBankAccountStatus
+  @objc public private(set) var status: STPBankAccountStatus = .new
```

Comment:
optionality for these is wrong. Should mirror checks in the decode function

Reply:
This seems right to me: In Objective-C, the fingerprint is optional, the status string is required, but the status defaults to `.new` if the string can't be mapped to a valid status. The decode function reflects that in Swift and Objective-C.